jax/_src/random.py

```diff
@@ -488,6 +488,8 @@ def choice(key: KeyArray,
       r = p_cuml[-1] * (1 - uniform(key, shape))
       ind = jnp.searchsorted(p_cuml, r)
     else:
+      if np.count_nonzero(p) < n_draws:
```

Comment:
I'm afraid this won't work in JAX in general: if this function is under a `jit` decorator, as it frequently will be, you cannot test the concrete values since at trace time you will only have abstract values.

We're working on adding some support for traced errors, though.

@LenaMartens

Reply:
Thanks for the response @hawkinsp, still trying to grasp some of the basics of Jax and this definitely helps. Should I close this PR?